src/d3a/models/strategy/future/strategy.py

```diff
@@ -0,0 +1,253 @@
+
+    def update(self, market: "FutureMarkets", strategy: "BidEnabledStrategy") -> None:
+        """Update the price of existing bids to reflect the new rates."""
+        for time_slot in strategy.area.future_markets.market_time_slots:
+            if self.time_for_price_update(strategy, time_slot):
+                if strategy.are_bids_posted(market.id, time_slot):
+                    strategy.update_bid_rates(market, self.get_updated_rate(time_slot))
+
+
+class FutureTemplateStrategyOfferUpdater(TemplateStrategyOfferUpdater):
+    """Version of TemplateStrategyOfferUpdater class for future markets"""
+
+    @property
+    def _time_slot_duration_in_seconds(self) -> int:
+        return GlobalConfig.FUTURE_MARKET_DURATION_HOURS * 60 * 60
+
+    @staticmethod
+    def get_all_markets(area: "Area") -> List["FutureMarkets"]:
+        """Override to return list of future markets"""
+        return [area.future_markets]
+
+    @staticmethod
+    def get_all_time_slots(area: "Area") -> List[DateTime]:
+        """Override to return all future market available time slots"""
+        return area.future_markets.market_time_slots
+
+    def update(self, market: "FutureMarkets", strategy: "BidEnabledStrategy") -> None:
+        """Update the price of existing offers to reflect the new rates."""
+        for time_slot in strategy.area.future_markets.market_time_slots:
+            if self.time_for_price_update(strategy, time_slot):
+                if strategy.are_offers_posted(market.id):
+                    strategy.update_offer_rates(market, self.get_updated_rate(time_slot))
+
+
+class FutureMarketStrategyInterface:
+    """Dummy/empty class that does not provide concrete implementation of the methods.
+    Is needed in order to disable the implementation of the future market strategy
+    when future markets are disabled by configuration."""
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def event_market_cycle(self, strategy: "BidEnabledStrategy") -> None:
+        """Base class method for handling the market cycle"""
+
+    def event_tick(self, strategy: "BidEnabledStrategy") -> None:
+        """Base class method for handling the tick"""
+
+
+class FutureMarketStrategy(FutureMarketStrategyInterface):
+    """Manages bid/offer trading strategy for the future markets, for a single asset."""
+    def __init__(self,
+                 initial_buying_rate: float, final_buying_rate: float,
+                 initial_selling_rate: float, final_selling_rate: float):
+        """
+        Args:
+            initial_buying_rate: Initial rate of the future bids
+            final_buying_rate: Final rate of the future bids
+            initial_selling_rate: Initial rate of the future offers
+            final_selling_rate: Final rate of the future offers
+        """
+        super().__init__()
+
+        self._update_interval = FutureTemplateStrategiesConstants.UPDATE_INTERVAL_MIN
+        self._bid_updater = FutureTemplateStrategyBidUpdater(
```

Comment:
Hmmm, I think this is better dealt with a new Interface class + a factory method. Reason for that is because if I use if / else branching, I would have to repeat this if / else on the event_tick method where the bid/offer updater are updating their clock / energy rates. Let me give it a try

Reply:
I know, I was also in between while implementing this. The only reason I did that was in order to potentially reuse it in template strategies in the future, for instance for the case of the storage strategy on one-sided market (where the bid_updater should not be constructed at all). But yeah, as you said, factory classes have unavoidable boilerplate.